components/tidb_query/src/rpn_expr/impl_arithmetic.rs

```diff
@@ -1253,4 +1274,90 @@ mod tests {
     fn str2dec(s: &str) -> Option<Decimal> {
         s.parse().ok()
     }
+
+    #[test]
+    fn test_real_divide() {
+        let normal = vec![
+            (double2real(2.2), double2real(1.1), double2real(2.0)),
+            (double2real(2.33), double2real(-0.01), double2real(-233.0)),
+            (double2real(2.33), double2real(0.01), double2real(233.0)),
+            (None, double2real(2.0), None),
+            (double2real(123.0), None, None),
+        ];
+
+        for (lhs, rhs, expected) in normal {
+            let actual = RpnFnScalarEvaluator::new()
+                .push_param(lhs.clone())
+                .push_param(rhs.clone())
+                .evaluate(ScalarFuncSig::DivideReal)
+                .unwrap();
+
+            assert_eq!(actual, expected, "lhs={:?}, rhs={:?}", lhs, rhs);
+        }
+
+        let abnormal = vec![
+            (double2real(2.33), double2real(0.0)),
+            (double2real(2.33), double2real(-0.0)),
+        ];
+
+        // Vec<[(Flag, SqlMode, is_ok(bool), has_warning(bool))]>
+        let modes = vec![
```

Comment:
These code looks very similar to `decimal_divide` and is very verbose. Please find a way to simplify it.

Reply:
OK. I will try.